src/app/tests/TestDefaultICDClientStorage.cpp

```diff
@@ -208,6 +212,43 @@ void TestClientInfoCountMultipleFabric(nlTestSuite * apSuite, void * apContext)
     NL_TEST_ASSERT(apSuite, count == 0);
 }
 
+void TestProcessCheckInPayload(nlTestSuite * apSuite, void * apContext)
+{
+    CHIP_ERROR err       = CHIP_NO_ERROR;
+    FabricIndex fabricId = 1;
+    NodeId nodeId        = 6666;
+    TestPersistentStorageDelegate clientInfoStorage;
+    TestSessionKeystoreImpl keystore;
+
+    DefaultICDClientStorage manager;
+    err = manager.Init(&clientInfoStorage, &keystore);
+    NL_TEST_ASSERT(apSuite, err == CHIP_NO_ERROR);
+    err = manager.UpdateFabricList(fabricId);
+    NL_TEST_ASSERT(apSuite, err == CHIP_NO_ERROR);
+    // Populate clientInfo
+    ICDClientInfo clientInfo1;
+    clientInfo1.peer_node = ScopedNodeId(nodeId, fabricId);
+
+    err = manager.SetKey(clientInfo1, ByteSpan(kKeyBuffer1));
+    NL_TEST_ASSERT(apSuite, err == CHIP_NO_ERROR);
+    err = manager.StoreEntry(clientInfo1);
+    NL_TEST_ASSERT(apSuite, err == CHIP_NO_ERROR);
+
+    uint32_t counter                  = 1;
+    System::PacketBufferHandle buffer = MessagePacketBuffer::New(chip::Protocols::SecureChannel::CheckinMessage::sMinPayloadSize);
+    MutableByteSpan output{ buffer->Start(), buffer->MaxDataLength() };
+    err = chip::Protocols::SecureChannel::CheckinMessage::GenerateCheckinMessagePayload(clientInfo1.shared_key, counter, ByteSpan(),
+                                                                                        output);
+    NL_TEST_ASSERT(apSuite, err == CHIP_NO_ERROR);
+
+    buffer->SetDataLength(static_cast<uint16_t>(output.size()));
+    ICDClientInfo clientInfo;
+    ByteSpan payload{ buffer->Start(), buffer->DataLength() };
+    bool refreshKey;
+    err = manager.ProcessCheckInPayload(payload, clientInfo, refreshKey);
+    NL_TEST_ASSERT(apSuite, err == CHIP_NO_ERROR);
```

Comment:
please also add test cases.
the key in Checkin message is different from the one in storage, then we expect the silence without error.
Please add test for VerifyOrReturnError(checkInCounter > clientInfo.offset, CHIP_ERROR_INVALID_SIGNATURE);
Please add test for checkInCounter > kCheckInRolloverConstant

Reply:
I'm not sure the tests should go here. Counter and Payload corruption tests should go in the CheckInMessage tests since thats what actually processes the message.

TestCheckInMessage test are several lacking right now (not this PRs doing). this can be a follow up since that their is a bug in the tests and missing steps in the processing.